docs/editors/overview.md

```diff
@@ -38,181 +39,207 @@ functionality.
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">❌</td>
+    <td align="center">✅</td>
   </tr>
   <tr>
     <td>Parameter hints</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">❌</td>
   </tr>
   <tr>
     <td>Find references</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">✅</td>
   </tr>
   <tr>
     <td>Run/Debug</td>
     <td align="center">✅</td>
     <td align="center"></td>
     <td align="center"></td>
     <td align="center">✅</td>
+    <td align="center">❌</td>
   </tr>
   <tr>
     <td>Find implementations</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">✅</td>
   </tr>
   <tr>
     <td>Rename symbol</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">✅</td>
   </tr>
   <tr>
     <td>Code actions</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">✅</td>
   </tr>
   <tr>
     <td>Worksheets</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">❌</td>
   </tr>
   <tr>
     <td>Document symbols</td>
     <td align="center">✅</td>
     <td align="center">Flat</td>
     <td align="center">Flat</td>
     <td align="center">✅</td>
+    <td align="center">Flat</td>
   </tr>
   <tr>
     <td>Workspace symbols</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">❌</td>
   </tr>
   <tr>
     <td>Formatting</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">✅</td>
   </tr>
   <tr>
     <td>Folding</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center"> </td>
     <td align="center">✅</td>
+    <td align="center">❌</td>
   </tr>
   <tr>
     <td>Highlight</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
+    <td align="center">❌</td>
   </tr>
   <tr>
     <td>Metals Extensions</td>
     <td align="center">✅</td>
     <td align="center">✅</td>
     <td align="center">Status bar, Input box, Decoration protocol, Did focus</td>
     <td align="center">✅</td>
+    <td align="center">❌</td>
   </tr>
   <tr>
      <td>Organize imports</td>
      <td align="center">✅</td>
      <td align="center">✅</td>
      <td align="center">✅</td>
      <td align="center">✅</td>
+     <td align="center">❌</td>
```

Comment:
If I understand it correctly, Metals provides a custom command (non-workspace one) to organize imports. Wonder if it would be possible for Metals to expose it through a code action as well, LSP even appears to define a special code action kind for such operation. But then again, I'm not sure how exactly this works. :smile:

Reply:
This should work with the default one. No fully sure how the Helix puts it together, but for example in nvim-metals when we trigger an organize imports it ends up looking like this when sent to the server.

```
[Trace - 09:04:43 AM] Received request 'textDocument/codeAction - (10)'
Params: {
  "textDocument": {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/minimal/project.scala"
  },
  "range": {
    "start": {
      "line": 4,
      "character": 1
    },
    "end": {
      "line": 4,
      "character": 1
    }
  },
  "context": {
    "diagnostics": [],
    "only": [
      "source.organizeImports"
    ]
  }
}
```
The only thing that really matters here is the `context` and the `textDocument`.  I do put it together though client side [here](https://github.com/scalameta/nvim-metals/blob/d90ea43ded7ec651606b0a533ae9740083436e58/lua/metals.lua#L307-L330) before it gets sent.

Reply:
Why is the organize imports actions only suggested when `context.only` contains `source.organizeImports`? Helix does not currently (to my understanding) allow any custom logic in LSP communication, which means that it is impossible to include this `context.only` thing.

Would it be possible offer `source.organizeImports` action when the range in  `textDocument/codeAction` contains imports or the whole file is selected?

Reply:
> EDIT: Btw what is the difference between "Fix imports" quick fix and source.organizeImports?

Quick fix only pops up when you get unused warnings, the other one is invoked usually manually.

> Why is the organize imports actions only suggested when context.only contains source.organizeImports? Helix does not currently (to my understanding) allow any custom logic in LSP communication, which means that it is impossible to include this context.only thing.

Organize imports runs scalafix, so we don't want to invoke it other than when it's done on purpose.

Reply:
@tgodzik Would it be an option to make it so that Metals advertises the "organize imports" code action whenever the cursor or selection contains any imports, without any special support required from the client (I think that's the same thing that @jpaju has proposed above)? Or is there some limitation here or maybe I misunderstood how things work? :slightly_smiling_face:

Reply:
We could, but we would have to create a lazy organize imports rule since the action will invoke scalafix, which would mean that whenever imports are selected, cursor moved around them or code being written, we would again invoke scalafix as actions are recalculated continuously

Reply:
Thanks for the explanation, I'll try to open a feature request later. :+1:

Reply:
We could actually do it quite simply. We can just add the command for the code action to run a single rule, which would be organize imports. The command already exists and an arbitrary command can be attached to a code action. In this case we would never invoke scalafix unless the action is run.

We would do that instead of running in the action itself. I would modify https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala#L123 we could make it a totally separate action and allow to run it always when in imports.

We could also play a bit in the code to also change the main organize imports to use the command, but that's not neccessary.